internal/helm/registry/client.go

```diff
@@ -35,9 +36,16 @@ func ClientGenerator(isLogin bool) (*registry.Client, string, error) {
 			return nil, "", err
 		}
 
+		var errs []error
 		rClient, err := registry.NewClient(registry.ClientOptWriter(io.Discard), registry.ClientOptCredentialsFile(credentialFile.Name()))
 		if err != nil {
-			return nil, "", err
+			errs = append(errs, err)
+			// attempt to delete the temporary file
+			err := os.Remove(credentialFile.Name())
```

Comment:
Based on similar code in the reconcilers where the secret file is deleted, here we aren't checking if the file is empty. Wondering if we should add a check in the file name or use `os.RemoveAll()` instead just to avoid error due to failure to delete file that doesn't exist.